var/spack/repos/builtin/packages/bamutil/package.py

```diff
@@ -36,6 +36,8 @@ class Bamutil(MakefilePackage):
 
     version('1.0.13', '08b7d0bb1d60be104a11f0e54ddf4a79')
 
+    depends_on('zlib', type=('build', 'link'))
```

Comment:
Technically this is the default, so you don't need to specify the `type`. But since `zlib` doesn't have any binaries in `prefix.bin`, `type='link'` is technically more correct.